### PR TITLE
Select agents and add/remove groups

### PR DESCRIPTION
**Pull request by lucianogorza**
### Description
This PR adds global actions to the agent's table to allow the user to add agents to groups or remove agents from groups.
To achieve this, a selector was added to the table, and the TableWzAPI component had to be modified. Global actions to edit agent groups open a modal from which groups to add or remove can be selected. Upon saving, the progress of the process is displayed, first retrieving agent information and then applying changes to each group.
Finally, the result of the transactions is shown, providing information to the user about agents that were successfully assigned or removed from groups and indicating the cause of any failures if they occurred.

In summary, the following modifications were made:
- Add global actions to agent's table
  - Add agents to groups
  - Remove agents from groups
- Add "addOnTitle" optional prop to `TableWzAPI` component to add selection options after the title on the table.
- Enhance the `TableWzAPI` header for better visibility across various screen widths.
- Utilize the `ref` prop in `TableWithSearchBar` to reset the selection when the table is refreshed, sorted, or the page is changed.
- Enhance the `apiReq` method in the `WzRequest` class by introducing an optional parameter `returnOriginalResponse`. This parameter allows the method to return all affected and failed items when adding agents to groups or removing agents from groups.

### Issues Resolved
https://github.com/wazuh/wazuh-dashboard-plugins/issues/6257

#### Select some agents
image
image
image

#### Select all filtered agents
image
image
image
image
image

### Test
Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

### General

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Without select agents global actions must be disabled | :black_circle: | :black_circle: | :black_circle: |
| Select some agents. Global actions must be enabled | :black_circle: | :black_circle: | :black_circle: |
| Select agents and change filters. Selection must cleared | :black_circle: | :black_circle: | :black_circle: |
| Select agents and change page. Selection must cleared | :black_circle: | :black_circle: | :black_circle: |
| Select agents and sort data. Selection must cleared | :black_circle: | :black_circle: | :black_circle: |
| With a read only user, select some agents. Global actions must be disabled | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Without select agents global actions must be disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select some agents. Global actions must be enabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select agents and change filters. Selection must cleared</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select agents and change page. Selection must cleared</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select agents and sort data. Selection must cleared</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With a read only user, select some agents. Global actions must be disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Select some agents

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different agents and groups. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Remove groups from agents. Select groups, save, and check the results. Test with different agents and groups. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different agents and groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Add/Remove groups -> Remove groups from agents. Select groups, save, and check the results. Test with different agents and groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Select all filtered agents

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Add/Remove groups -> Add groups to agents. A callout should be displayed warning that the changes will be applied to all filtered agents. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different groups. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Remove groups from agents. A callout should be displayed warning that the changes will be applied to all filtered agents. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Remove groups from agents. Select groups, save, and check the results. Test with different groups. | :black_circle: | :black_circle: | :black_circle: |

**Details**

<details>
<summary>:black_circle: Go to Add/Remove groups -> Add groups to agents. A callout should be displayed warning that the changes will be applied to all filtered agents</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Add/Remove groups -> Remove groups from agents. A callout should be displayed warning that the changes will be applied to all filtered agents</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Remove groups from agents. Select groups, save, and check the results. Test with different groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Paginate requests

The following request must be paginated:
- Agents list: When adding or removing groups for all filtered agents, the agents must be retrieved through paginated requests.
- Groups list: When the modal is open, the list of groups is retrieved.
- Modify groups: When selecting groups and saving, the changes must be applied to the agents through paginated requests.

The default pagination is set to 1000. To test, it is necessary to modify the default value, for example, to 2.

- `plugins/main/public/components/endpoints-summary/services/get-agents.tsx`: Line 9.
- `plugins/main/public/components/endpoints-summary/services/get-groups.tsx`: Line 9.
- `plugins/main/public/components/endpoints-summary/services/paginated-agents-group.tsx`: Line 17.

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Select agents > Go to Add/Remove groups -> Add groups to agents. Check the groups list and the network requests. | :black_circle: | :black_circle: | :black_circle: |
| Select all filtered agents > Go to Add/Remove groups -> Add groups to agents. Select groups and save. Check the network requests to get the agents and add the groups. Check the displayed result. | :black_circle: | :black_circle: | :black_circle: |

**Details**

<details>
<summary>:black_circle: Select agents > Go to Add/Remove groups -> Add groups to agents. Check the groups list and the network requests</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select all filtered agents > Go to Add/Remove groups -> Add groups to agents. Select groups and save. Check the network requests to get the agents and add the groups. Check the displayed result</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by Tostti** [comment]
## UI

### General

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Without select agents global actions must be disabled | :black_circle: | :black_circle: | :black_circle: |
| Select some agents. Global actions must be enabled | :black_circle: | :black_circle: | :black_circle: |
| Select agents and change filters. Selection must cleared | :black_circle: | :black_circle: | :black_circle: |
| Select agents and change page. Selection must cleared | :black_circle: | :black_circle: | :black_circle: |
| Select agents and sort data. Selection must cleared | :black_circle: | :black_circle: | :black_circle: |
| With a read only user, select some agents. Global actions must be disabled | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Without select agents global actions must be disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select some agents. Global actions must be enabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select agents and change filters. Selection must cleared</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select agents and change page. Selection must cleared</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select agents and sort data. Selection must cleared</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With a read only user, select some agents. Global actions must be disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Select some agents

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different agents and groups. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Remove groups from agents. Select groups, save, and check the results. Test with different agents and groups. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different agents and groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Add/Remove groups -> Remove groups from agents. Select groups, save, and check the results. Test with different agents and groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Select all filtered agents

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Add/Remove groups -> Add groups to agents. A callout should be displayed warning that the changes will be applied to all filtered agents. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different groups. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Remove groups from agents. A callout should be displayed warning that the changes will be applied to all filtered agents. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Remove groups from agents. Select groups, save, and check the results. Test with different groups. | :black_circle: | :black_circle: | :black_circle: |

**Details**

<details>
<summary>:black_circle: Go to Add/Remove groups -> Add groups to agents. A callout should be displayed warning that the changes will be applied to all filtered agents</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Add/Remove groups -> Remove groups from agents. A callout should be displayed warning that the changes will be applied to all filtered agents</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Remove groups from agents. Select groups, save, and check the results. Test with different groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Paginate requests

The following request must be paginated:
- Agents list: When adding or removing groups for all filtered agents, the agents must be retrieved through paginated requests.
- Groups list: When the modal is open, the list of groups is retrieved.
- Modify groups: When selecting groups and saving, the changes must be applied to the agents through paginated requests.

The default pagination is set to 1000. To test, it is necessary to modify the default value, for example, to 2.

- `plugins/main/public/components/endpoints-summary/services/get-agents.tsx`: Line 9.
- `plugins/main/public/components/endpoints-summary/services/get-groups.tsx`: Line 9.
- `plugins/main/public/components/endpoints-summary/services/paginated-agents-group.tsx`: Line 17.

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Select agents > Go to Add/Remove groups -> Add groups to agents. Check the groups list and the network requests. | :black_circle: | :black_circle: | :black_circle: |
| Select all filtered agents > Go to Add/Remove groups -> Add groups to agents. Select groups and save. Check the network requests to get the agents and add the groups. Check the displayed result. | :black_circle: | :black_circle: | :black_circle: |

**Details**

<details>
<summary>:black_circle: Select agents > Go to Add/Remove groups -> Add groups to agents. Check the groups list and the network requests</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select all filtered agents > Go to Add/Remove groups -> Add groups to agents. Select groups and save. Check the network requests to get the agents and add the groups. Check the displayed result</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

File: plugins/main/public/components/endpoints-summary/services/paginated-agents-group.test.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
+// Importa las funciones y tipos necesarios
```
Comment (yenienserrano) [inline plugins/main/public/components/endpoints-summary/services/paginated-agents-group.test.tsx:1]: Check this comment 👍🏼

**Comment by jbiset** [comment]
## UI

### General

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Without select agents global actions must be disabled | :black_circle: | :black_circle: | :black_circle: |
| Select some agents. Global actions must be enabled | :black_circle: | :black_circle: | :black_circle: |
| Select agents and change filters. Selection must cleared | :black_circle: | :black_circle: | :black_circle: |
| Select agents and change page. Selection must cleared | :black_circle: | :black_circle: | :black_circle: |
| Select agents and sort data. Selection must cleared | :black_circle: | :black_circle: | :black_circle: |
| With a read only user, select some agents. Global actions must be disabled | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Without select agents global actions must be disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select some agents. Global actions must be enabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select agents and change filters. Selection must cleared</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select agents and change page. Selection must cleared</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select agents and sort data. Selection must cleared</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With a read only user, select some agents. Global actions must be disabled</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Select some agents

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different agents and groups. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Remove groups from agents. Select groups, save, and check the results. Test with different agents and groups. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different agents and groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Add/Remove groups -> Remove groups from agents. Select groups, save, and check the results. Test with different agents and groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Select all filtered agents

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Go to Add/Remove groups -> Add groups to agents. A callout should be displayed warning that the changes will be applied to all filtered agents. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different groups. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Remove groups from agents. A callout should be displayed warning that the changes will be applied to all filtered agents. | :black_circle: | :black_circle: | :black_circle: |
| Go to Add/Remove groups -> Remove groups from agents. Select groups, save, and check the results. Test with different groups. | :black_circle: | :black_circle: | :black_circle: |

**Details**

<details>
<summary>:black_circle: Go to Add/Remove groups -> Add groups to agents. A callout should be displayed warning that the changes will be applied to all filtered agents</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Add/Remove groups -> Add groups to agents. Select groups, save, and check the results. Test with different groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Go to Add/Remove groups -> Remove groups from agents. A callout should be displayed warning that the changes will be applied to all filtered agents</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Remove groups from agents. Select groups, save, and check the results. Test with different groups</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Paginate requests

The following request must be paginated:
- Agents list: When adding or removing groups for all filtered agents, the agents must be retrieved through paginated requests.
- Groups list: When the modal is open, the list of groups is retrieved.
- Modify groups: When selecting groups and saving, the changes must be applied to the agents through paginated requests.

The default pagination is set to 1000. To test, it is necessary to modify the default value, for example, to 2.

- `plugins/main/public/components/endpoints-summary/services/get-agents.tsx`: Line 9.
- `plugins/main/public/components/endpoints-summary/services/get-groups.tsx`: Line 9.
- `plugins/main/public/components/endpoints-summary/services/paginated-agents-group.tsx`: Line 17.

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Select agents > Go to Add/Remove groups -> Add groups to agents. Check the groups list and the network requests. | :black_circle: | :black_circle: | :black_circle: |
| Select all filtered agents > Go to Add/Remove groups -> Add groups to agents. Select groups and save. Check the network requests to get the agents and add the groups. Check the displayed result. | :black_circle: | :black_circle: | :black_circle: |

**Details**

<details>
<summary>:black_circle: Select agents > Go to Add/Remove groups -> Add groups to agents. Check the groups list and the network requests</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Select all filtered agents > Go to Add/Remove groups -> Add groups to agents. Select groups and save. Check the network requests to get the agents and add the groups. Check the displayed result</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

File: plugins/main/public/components/endpoints-summary/services/paginated-agents-group.test.tsx (hunk 1 above)
Comment (lucianogorza) [reply plugins/main/public/components/endpoints-summary/services/paginated-agents-group.test.tsx]: Removed

**Review by Tostti — APPROVED** [review]
Test ✅ 
CR ✅ 

LGTM

**Review by jbiset — APPROVED** [review]
## CR :green_circle: 

## Test :green_circle:

**Pull request merged**